dashboard/app/models/pd/post_course_survey.rb

```diff
@@ -13,12 +13,82 @@
 # Indexes
 #
 #  index_pd_post_course_surveys_on_form_id                (form_id)
+#  index_pd_post_course_surveys_on_form_user_year_course  (form_id,user_id,year,course) UNIQUE
 #  index_pd_post_course_surveys_on_submission_id          (submission_id) UNIQUE
 #  index_pd_post_course_surveys_on_user_form_year_course  (user_id,form_id,year,course) UNIQUE
 #  index_pd_post_course_surveys_on_user_id                (user_id)
 #
 
 module Pd
   class PostCourseSurvey < ActiveRecord::Base
+    include JotFormBackedForm
+
+    VALID_YEARS = [
+      YEAR_18_19 = '2018-2019'.freeze
+    ].freeze
+
+    VALID_COURSES = [
+      COURSE_CSD = 'CS Discoveries'.freeze,
+      COURSE_CSP = 'CS Principles'.freeze
+    ].freeze
+
+    CURRENT_YEAR = YEAR_18_19
+
+    belongs_to :user
+
+    # @override
+    def self.attribute_mapping
+      {
+        user_id: 'userId',
+        course: 'csp_or_csd'
+      }
+    end
+
+    validates_uniqueness_of :user_id, scope: [:form_id, :year, :course],
+      message: 'already has a submission for this form, year, and course'
+
+    validates_presence_of(
+      :user_id,
```

Comment:
form_id and submission_id belong here as well

Reply:
They are inherited: https://github.com/code-dot-org/code-dot-org/pull/22824/files#diff-526619cced417df2c7f67bd4593661baR14